includes/functions.php

```diff
@@ -30,16 +30,22 @@ function get_directory_blog_id () {
 
 if (!function_exists('get_site_directory_terms')) :
     /**
-     * Gets all categories in the site directory.
+     * Gets categories in the site directory.
+     *
+     * @param array $slugs Optional. Slugs of categories to retrieve. Default is to retrieve all categories.
+     * @param array $exclude Optional. IDs of categories to exclude.
      *
      * @uses get_terms()
      *
      * @return array|false|WP_Error
      */
-    function get_site_directory_terms () {
+    function get_site_directory_terms ($slugs = array(), $exclude = array() ) {
```

Comment:
This is needlessly limiting. Instead of re-writing the same *implementation* of `get_terms()`, mimic its *interface* so that callers can use this function transparently and intuitively. In other words, instead of defining parameters for each argument you later pass into the [core `get_terms()` function](https://developer.wordpress.org/reference/functions/get_terms/) one at a time, use the same function signature as the core function:

```php
function get_site_directory_terms ($args = array()) { // notice the same function signature as the core function
    // …
    $terms = get_terms($args); // now you can transparently pass $args through, and let core handle it
    // …
}
```

This way you have less code and more functionality at the same time.